### PR TITLE
[IPCT1-582] - Calculate ssi

### PR DESCRIPTION
**Pull request by dev-jotape**
This PR fixes [IPCT1-582] at https://impactmarket.atlassian.net/browse/IPCT1-582

## Changes
Change `calcuateCommunitiesMetrics` to use subgraph to calculate ssi

### CODE REVIEW

File: packages/core/src/interfaces/ubi/beneficiary.ts [hunk 1]
```diff
@@ -86,3 +86,12 @@ export interface BeneficiaryCreationAttributes {
...
 }
+
+export interface BeneficiarySubgraph {
+    claims: number;
+    community: {
+        id: string;
+    };
+    lastClaimAt: number;
+    preLastClaimAt: number;
+}
```
Comment (obernardovieira) [inline packages/core/src/interfaces/ubi/beneficiary.ts:97]: maybe I would keep this on a separate file, where, later, might have all TheGraph entities types.

File: packages/core/src/subgraph/config.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import 'cross-fetch/polyfill';
```
Comment (obernardovieira) [inline packages/core/src/subgraph/config.ts:1]: what is cross-fetch for?

File: packages/core/src/subgraph/queries/beneficiary.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { gql } from 'apollo-boost';
+
+import { BeneficiarySubgraph } from '../../interfaces/ubi/beneficiary';
+import { client } from '../config';
+
+export const getBeneficiaries = async (
+    communities: string[]
+): Promise<BeneficiarySubgraph[]> => {
```
Comment (obernardovieira) [inline packages/core/src/subgraph/queries/beneficiary.ts:8]: I'm afraid that this end up consuming all the RAM possible

File: packages/core/src/subgraph/config.ts (hunk 2 above)
Comment (dev-jotape) [reply packages/core/src/subgraph/config.ts:1]: I was having some problems for run the Apollo Client in node.js (without a browser) and found the answer here: https://stackoverflow.com/questions/50688998/using-apolloclient-with-node-js-fetch-is-not-found-globally-and-no-fetcher-pas

basically, to use the apollo client you need to polyfill or provide a fetch method

Comment (obernardovieira) [reply packages/core/src/subgraph/config.ts:1]: :+1: 
add a comment with that, please

File: packages/core/src/interfaces/ubi/beneficiary.ts (hunk 1 above)
Comment (dev-jotape) [reply packages/core/src/interfaces/ubi/beneficiary.ts]: right, I change to `src/subgraph/interfaces` to be with everything related to subgraph

File: packages/core/src/subgraph/queries/beneficiary.ts (hunk 3 above)
Comment (dev-jotape) [reply packages/core/src/subgraph/queries/beneficiary.ts:8]: added batchs

**Review by obernardovieira — COMMENTED** [review]
Last changes make them all run sequential, or did I misunderstand it?
The goal is to process a batch in parallel. Eg: 20 being processed at the same time, once they all finish, then, another 20.
I also think it's better if `getBeneficiaries` queries only one community. Simplifies code readability, the query and remove some extra processing

**Comment by dev-jotape** [comment]
you were right, I changed the code to use more promises and process 20 communities at a time.
could you validate again?

**Review by obernardovieira — APPROVED** [review]

**Pull request merged**